### ISSUE TITLE
Gaphor diagrams in Sphinx generated documentation

### ISSUE THREAD

**Issue by amolenaar** [gaphor/gaphor#819]
## Is your feature request related to a problem?

We use Sphinx to create the tech docs for Gaphor. It would be nice if the model diagrams could incorporated in the docs in an easy way.

Currently we generate the diagrams and check them in to version control. it would be nice if that's no longer necessary.

## Describe the solution you'd like

It would be nice to have a little Sphinx plugin that generates diagrams on demand. Use a syntax like:

```rst
.. gaphor::
   :model: path/to/my-model.gaphor
   :diagram: root.my-diagram
   :caption: provide a nice caption for the diagram
```

Not sure if the model should be defined in every `gaphor` element

## Describe alternatives you've considered

We can stick with the old scripts, but I'm afraid that that solution will not scale.
The old scripts also generate the pages. Not sure if this plugin should also do that.

## Additional context

The solution should work with readthedocs.io as well. I'm not sure to what extend diagrams can be rendered over there or if the diagrams need to be pre-rendered.

**Comment by amolenaar** [comment]
It looks like cairo-dev is available on RtD: https://github.com/readthedocs/readthedocs.org/issues/1037. That would mean we can install pycairo. This should be enough for us to render the models.

**Comment by danyeaw** [comment]
Closed by #1134.
